### DISCUSSION TITLE
How to change the idle connection timeout in axum/hyper?

### DISCUSSION THREAD

**Discussion by kobe2000** [tokio-rs/axum#2938] [Q&A]
### Summary

I keep getting 502 http code occasionally when I put an axum application behind treafik, I guess that it may because of the timeout for idle connection. (aka keepalive_timeout of nginx)

I noticed that hyper has an [header_read_timeout](https://docs.rs/hyper/latest/hyper/server/conn/http1/struct.Builder.html#method.header_read_timeout) config which has default value of 30 seconds. I have no idea if it is effective and how to set it under axum.

Please help, thank you.

### axum version

0.7.6

**Comment by jplatte** [comment]
See #2939, which is for another hyper server setting. The same answer applies here.

**Comment by gmarti** [comment]
What about https://github.com/tokio-rs/axum/pull/3852
